python/example_code/apigateway/aws_service/README.md

```diff
@@ -1,86 +1,129 @@
-long to fit within an HTTPS GET request, it may be truncated or generate a `414: URI Too Long`
-error. In such cases, it is better to either parse and translate sentences individually or pass
-the text in the body of a POST request. Implementation of a POST interface to the TranslateText
-method is left as an exercise for the reader.
+This example requires a DynamoDB table with a specific key schema and an 
+AWS Identity and Access Management (IAM) role that grants permission to let
+API Gateway perform actions on the table. These resources are managed by
+an AWS CloudFormation stack that is defined in the accompanying `setup.yaml` file. 
 
-Note: Text with embedded newlines is currently not supported by Amazon Translate. Such text results 
-in a Serialization Exception.
+### Deploy resources
 
-## Repository files
+Deploy prerequisite resources by running the example script with the `deploy` flag at 
+a command prompt.
 
-* `aws_service.py` : Main program source file
-* `Tale_of_Two_Cities.txt` : Sample text to translate 
+```
+py aws_service.py deploy
+```
 
-## AWS infrastructure resources
+### Run the usage demonstration
 
-* API Gateway REST API to the Amazon Translate TranslateText method
-* API Gateway usage plan
-* API Gateway API key
-* AWS Identity and Access Management (IAM) role for the API Gateway resource
+Run the usage example with the `demo` flag at a command prompt.
 
-## Prerequisites
+```
+python aws_service.py demo
+``` 
+
+### Destroy resources
+
+Destroy example resources by running the script with the `destroy` flag at a command 
+prompt.
+
+```
+python aws_service.py destroy
+``` 
+
+### Example structure
+
+The example contains the following files.
+
+**aws_service.py**
+
+Shows how to use Amazon API Gateway to create a REST API that is backed by a DynamoDB
+database. The example uses API Gateway integration with DynamoDB, so after the REST API
+is created and deployed, no user code is needed to serve the REST API. The `usage_demo`
+script in this file shows how to accomplish the following actions:
+
+1. Create a REST API served by API Gateway.
+1. Add resources to the REST API that represent a user profile.
+1. Add integration methods so the REST API uses a DynamoDB table to store user profile
+data. 
+1. Use the Requests package to call the REST API to add and retrieve user profiles.
+
+**setup.yaml**
+
+Contains a CloudFormation script that can be used to create the resources needed for 
+the demo. 
+
+*Resources*
+
+* A **DynamoDB table** that contains user profile information
+* An **IAM role** that grants API Gateway permission to read from and write to the 
+table.
+
+*Deploy and destroy*
+
+* Pass the `deploy` or `destroy` flag to the `aws_service.py` script.
+* Or, use the [AWS Command Line Interface (AWS CLI)](https://docs.aws.amazon.com/cli/) 
+[cloudformation](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/cloudformation/index.html#cli-aws-cloudformation) 
+commands.
```

Comment:
Might an example of running the AWS CLI commands be helpful here?

Reply:
I went the other way: because the python code can deploy/destroy with CloudFormation I think CLI info is unnecessary, so I removed mention of the CLI entirely.